xarray/tests/test_plot.py

```diff
@@ -313,12 +313,11 @@ def test_2d_coords_line_plot(self):
             coords={"lat": (("y", "x"), lat), "lon": (("y", "x"), lon)},
         )
 
-        hdl = da.plot.line(x="lon", hue="x")
-        assert len(hdl) == 5
-
-        plt.clf()
-        hdl = da.plot.line(x="lon", hue="y")
-        assert len(hdl) == 4
+        with figure_context():
+            hdl = da.plot.line(x="lon", hue="x")
+            assert len(hdl) == 5
+            hdl = da.plot.line(x="lon", hue="y")
```

Comment:
Thanks for the suggestion! Just to double-check, you want to wrap the two `plot-assert` into two different `figure_context()`, right? 
```
with figure_context():  
            hdl = da.plot.line(x="lon", hue="x")  
            assert len(hdl) == 5  
with figure_context():  
            hdl = da.plot.line(x="lon", hue="y")  
            assert len(hdl) == 4  
```

Reply:
Yes, correctly, such that the first figure gets closed before the second is created. For this test it is not strictly necessary because it works anyways. But it's certainly cleaner.